### PR TITLE
Closing some loose ends from previous front-end extension

### LINKED ISSUE #35: TxButton Should Check Sudo User
It makes sense if TxButton is set to "sudo" that it checks the `AccountPair` being passed to it is indeed the sudo user.

Since TxButton is component is reused in other components, and likely reused in components people create, it makes sense to add a bit more complexity/safety around it.

### LINKED ISSUE #73: Better error handling on invalid custom type
Front end can display an error message showing invalid custom type, as opposed to hanging on the loading screen.

### LINKED ISSUE #97: Allow decorating RPC information in the UI config

### LINKED ISSUE #98: Pallet Interactor supports Vec and Option parameters
Vec - a comma-separated list
Option - empty value to be submitted

### LINKED ISSUE #99: Unsubscription of transaction events in pallet interactor

### LINKED ISSUE #102: Deployed Instance is Not Functional
It appears that [the deployed instance](https://substrate.dev/substrate-front-end-template/) of the template is not working. I will remove it from the repo description in the meantime.

### PR DESCRIPTION
**Pull request by jimmychu0807**
This PR should:
- Fix #102 - Currently it is running substrate v2.0.0-rc2, so I removed the custom type needed in `production.env` for usual substrate-node-template. It is only applied when running in development mode (which is locally, and usually connecting to a node template)
- Fix #99 - unsubscribe to previous transaction events if a new transaction is submitted. Test by quickly submit two transactions. The finalized message of the previous tx should not display.
- Fix #98 - support Vec param (separated by `,`) and Option param (by leaving the field empty)
- Fix #97 - take custom decorating RPC and default polkadot RPC params
- Fix #73
- Fix #35 - check if sudo key doesn't match with current user key, sudo tx button is disabled. Test by changing the `AccountSelector` dropdown to anyone except Alice.

### CODE REVIEW

File: src/substrate-lib/components/DeveloperConsole.js [hunk 1]
```diff
@@ -1,12 +0,0 @@
...
-
-export default function DeveloperConsole (props) {
-  const { api } = useSubstrate();
-  window.api = api;
-  window.util = require('@polkadot/util');
-  window.util_crypto = require('@polkadot/util-crypto');
-  window.keyring = require('@polkadot/keyring');
-
-  return null;
-}
```
[the remark this answers is not in the document]
Comment (danforbes) [reply src/substrate-lib/components/DeveloperConsole.js:12]: Agree I think this is an important capability for the front-end template as a tool for learning :nerd_face:

[the remark this answers is not in the document]
Comment (jimmychu0807) [reply src/substrate-lib/components/DeveloperConsole.js:12]: Will revert it. I thought nobody is using it.

**Comment by jimmychu0807** [comment]
@JoshOrndorff 
> I wanted to test whether the production build works with the hosted node, but couldn't figure out how to get past this (output of yarn build):

> The project was built assuming it is hosted at /substrate-front-end-template/.
You can control this with the homepage field in your package.json.

The production build is the one in master version. So the latest version is never deployed until it is merged to master.

To simulate and test for production build, replace everything in `src/config/development.json` with `src/config/production.json`, and run the front-end template locally.

**Comment by jimmychu0807** [comment]
> but I find this UX confusing

I am thinking of adding a small text message below the input text box telling ppl it is an optional param. So by not entering a param, it will be set as null.

If the value is required, having an empty value in the input text box will make the submit button disabled.

**Comment by jimmychu0807** [comment]
> I found it a little confusing how many RPC entities existed that I couldn't actually call

I should be able to improve upon this.

**Comment by jimmychu0807** [comment]
@JoshOrndorff 

- This PR now fixes #35, #99 also.
- The `pallets/RPC` dropdown only show those that have at least one callables.
- Option and optional parameters are handled as image below:

Option param in transaction:

Screenshot 2020-06-07 at 20 13 59

Optional param in RPC:

Screenshot 2020-06-07 at 20 13 52

I won't be writing new features, major fixes in this PR. If you are also happy with the change, this PR is good to merge.

**Pull request merged**